marketplace/token.go

```diff
@@ -9,10 +9,20 @@ import (
 // BearerToken represents the bearer token sent by the marketplace, and includes the Unix timestamp of the time when
 // it expires.
 type BearerToken struct {
-	Expiration *int    `json:"expiration"`
+	Expiration *int64  `json:"expiration"`
 	Token      *string `json:"access_token"`
 }
 
+// MarshalBinary implements the "BinaryMarshaller" interface to easily marshal the struct when using the Redis client.
+func (bt BearerToken) MarshalBinary() (data []byte, err error) {
```

Comment:
Where this is used ?

Reply:
I think it's used here: https://github.com/RedHatInsights/sources-api-go/pull/57/files#diff-9f23ee963589d96d28d468ae7ce847cd2dcd6631c19b9d9fc4ce3be971f3cd3cR52

If the struct implements the `BinaryMarshaller` interface redis will use that method to marshal it (per the comment).

Reply:
Exactly. Actually, Redis requires us to implement that `BinaryMarshaller` interface to be able to cache the token.